components/inputs/input-fieldset.js

```diff
@@ -35,36 +41,40 @@ class InputFieldset extends PropertyRequiredMixin(InputInlineHelpMixin(SkeletonM
 	}
 
 	static get styles() {
-		return [ super.styles, inputLabelStyles, offscreenStyles,
+		return [ super.styles, heading4Styles, inputLabelStyles, offscreenStyles,
 			css`
 				:host {
 					display: block;
 				}
 				:host([hidden]) {
 					display: none;
 				}
+				legend.d2l-heading-4 {
+					margin-block-start: 0;
```

Comment:
@geurts let me know if you'd like to adjust the bottom margin or leave it `1.5rem`.